### PR TITLE
feat: Add collection insights incomplete message

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [CX-2597]

### Description

Adds message to the My Collection Insights tab that indicates that not all artworks have insights.
Simulator Screen Shot - iPhone 12 Pro Max - 2022-06-01 at 16 10 13

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add My Collection insights incomplete message - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-2597]: https://artsyproduct.atlassian.net/browse/CX-2597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 1]
```diff
@@ -177,7 +198,7 @@ const MyCollection: React.FC<{
       />
 
       <StickyTabPageScrollView
-        contentContainerStyle={{ flexGrow: 1, justifyContent: "center" }}
+        contentContainerStyle={{ paddingBottom: space(2) }}
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/MyCollection.tsx:201]: @MounirDhahri Needed to revert this change because it messes with the layout when you only have two artworks in My Collection
Simulator Screen Shot - iPhone 12 Pro Max - 2022-06-01 at 15 59 42
.

File: src/app/Scenes/MyCollection/MyCollection.tsx [hunk 2]
```diff
@@ -90,8 +96,23 @@ const MyCollection: React.FC<{
...
+    // in order to show the message
+    // TODO: wait exactly until all artworks have been loaded
+    setTimeout(() => {
+      const artworksWithoutInsight = artworks.find((artwork) => !artwork._marketPriceInsights)
+
+      refreshMyCollectionInsights({
+        collectionHasArtworksWithoutInsights: !!(artworks.length && artworksWithoutInsight),
+      })
+    }, 3000)
+  }
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/MyCollection.tsx:110]: I needed a way to get the info about artworks without insights from the **My Collection** tab to the **Insights** tab. It works but we should find a smarter way to do that.

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx [hunk 3]
```diff
@@ -31,12 +38,20 @@ export const MyCollectionInsights: React.FC<{}> = ({}) => {
   const hasMarketSignals = !!data.me?.auctionResults?.totalCount
 
   useEffect(() => {
-    RefreshEvents.addListener(MY_COLLECTION_REFRESH_KEY, refresh)
+    RefreshEvents.addListener(MY_COLLECTION_REFRESH_KEY, handleRefreshEvent)
+    RefreshEvents.addListener(MY_COLLECTION_INSIGHTS_REFRESH_KEY, handleRefreshEvent)
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:42]: We need an extra My Collection Insight refresh listener to update the insights tab from the My Collection tab.

File: src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx [hunk 4]
```diff
@@ -31,12 +38,20 @@ export const MyCollectionInsights: React.FC<{}> = ({}) => {
...
-      RefreshEvents.removeListener(MY_COLLECTION_REFRESH_KEY, refresh)
+      RefreshEvents.removeListener(MY_COLLECTION_REFRESH_KEY, handleRefreshEvent)
+      RefreshEvents.removeListener(MY_COLLECTION_INSIGHTS_REFRESH_KEY, handleRefreshEvent)
     }
   }, [])
 
+  const handleRefreshEvent = (...args: any[]) => {
+    refresh()
+
+    setAreInsightsIncomplete(args[0].collectionHasArtworksWithoutInsights)
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/MyCollectionInsights.tsx:52]: Sending the info about artworks without collection with the event.

**Comment by MounirDhahri** [comment]
I am getting this error message when adding an artwork without insights
Simulator Screen Shot - iPhone 13 - 2022-06-01 at 17 21 18

**Comment by MounirDhahri** [comment]
niice thanks!

**Review by MounirDhahri — APPROVED** [review]
